next/.eslintrc.js

```diff
@@ -22,6 +22,14 @@ module.exports = {
   ],
   rules: {
     "prettier/prettier": "error",
+    "@typescript-eslint/no-misused-promises": [
```

Comment:
@joshua-scott  I found this discussion on hook_react_form's repository: https://github.com/react-hook-form/react-hook-form/discussions/8622 

It seems that it's accepted there that this is the recommended way to fix the issue of "Promise-returning function provided to attribute where a void return was expected."  so I went with this.